### PR TITLE
Alternative FFmpeg implementation of parsing header and Covert art with Taglib

### PR DESCRIPTION
**Pull request by illuusio**
This one adds general Mixxx Taglib support in FFmpeg/Avconv soundsource.

### CODE REVIEW

File: src/soundsourceffmpeg.cpp [hunk 1]
```diff
@@ -531,120 +540,252 @@ unsigned int SoundSourceFFmpeg::read(unsigned long size,
...
+    } else if (is_wav) {
+        TagLib::RIFF::WAV::File f(qBAFilename.constData());
+        if (!readFileHeader(this, f)) {
+            return ERR;
+        }
 
-#if LIBAVCODEC_VERSION_INT < 3622144
-    FmtCtx->max_analyze_duration = 999999999;
+        // Taglib 1.8.x doesn't provide an ID3v2Tag method for WAV files.
+#if TAGLIB_MAJOR_VERSION == 1 && TAGLIB_MINOR_VERSION == 8
```
Comment (daschuer) [inline src/soundsourceffmpeg.cpp]: this test should be changed like here https://github.com/mixxxdj/mixxx/commit/6a40409b0fa5bfb53c2d3e5e7b8ab055c748ac27

**Comment by illuusio** [comment]
Small fix to be inline with other soundsources

**Comment by kain88-de** [comment]
LGTM

Although by now I think it might be a good separate the TAG parsing from the soundsources.

**Comment by illuusio** [comment]
Please close this.. It's converted to global Mixxx system

**Pull request merged**